actix-rt/src/runtime.rs

```diff
@@ -84,7 +88,7 @@ impl Runtime {
     /// complete execution by calling `block_on` or `run`.
     pub fn block_on<F>(&mut self, f: F) -> F::Output
     where
-        F: Future + 'static,
+        F: Future,
```

Comment:
Can we get a test case set up to prevent regression.

Reply:
I have one test for this added in the tokio 0.3 pr. I would copy/paste it here.